docs: Migrate pre-proposed article from Ubuntu Wiki (#108)
canonical/kernel-docs: docs/how-to/testing-verification/test-kernel-in-proposed.md

    description: "Learn to enable -proposed pocket, install test kernels, and verify stability of kernels before release."
---

# How to test pre-release Ubuntu kernels

After a kernel's sources are prepared by the Canonical Kernel Team, they are
uploaded to public build PPAs for compilation and regression testing. Once

> built, you may download and test these kernels before migration to the -proposed

Reviewer: Hi @alicemunduruca !

I'm not sure what you mean here by "before migration to the -proposed" pocket.
Is this referring to `ppa:canonical-kernel-team/ppa`?
or the previous `pre-proposed`?

> pocket. It is worth mentionning however that adding these PPAs to a production

Reviewer: ```suggestion
pocket. It is worth mentioning however that adding these PPAs to a production
```

system could cause failures to boot or regressions, and so is heavily
discouraged.

Once testing and certification have concluded, the kernels are then copied to
the -proposed pocket for testing before being published to -updates and
-security. You can download these pre-release kernels to install and test them
before a stable release, and although they may still contain regressions, the

> kernels in -proposed are more stable as a result of the testing they undergo.

Reviewer: Hi @alicemunduruca !

I re-read the original guide and other sources and I have a better idea now.
crank -> Build PPA -> -proposed -> -updates.

I strongly suggest rewriting this intro but simplifying it and focus on the how-to.

````
Pre-release Ubuntu kernels are available in the Canonical Kernel Team (CKT)
build PPAs and `-proposed` pocket before they are published to `-updates`.
To install these pre-release kernels you need to opt in, as neither source is
enabled by default.

This guide explains how to enable and test pre-release Ubuntu kernels, and how
to report any regressions you find.

```{caution}
Kernels published to the build PPAs are not suitable for production systems -
enabling them could cause boot failures or regressions and is not recommended.
```

For details on the kernel lifecycle, see the <link to the explanation page here>.
````

Then expand the description for the kernel lifecycle into an Explanation page. See the attached file.

[kernel-lifecycle-sru.md](https://github.com/user-attachments/files/29325473/kernel-lifecycle-sru.md)

Author: Sounds good. I'll go ahead and start by adding everything except the link for the kernel lifecycle article.

Reviewer: Hi @alicemunduruca , i attached the Explanation file in my earlier comment (kernel-lifecycle-sru.md).
Did you manage to give it a look?
If it makes sense we can add it to this PR so it's a complete change.

````
`````

> ## Enable a CKT Build PPA

Reviewer: ```suggestion
## Enable the CKT build PPA
```

> You may add the build PPAs to your system as follows:

Reviewer: ```suggestion
You can add the CKT build PPAs to your system as follows:
```

```{code-block} none
sudo add-apt-repository ppa:canonical-kernel-team/ppa
sudo add-apt-repository ppa:canonical-kernel-team/ppa2
```

```{warning}
Do remember that these kernels may be entirely untested at time of
download and thus potentially fundamentally broken. Only test from build PPAs

> if you have a reason to do so.

Reviewer: ```suggestion
Do remember that these kernels may be entirely untested at time of download and
thus potentially fundamentally broken. Only test from build PPAs if you have a
reason to do so.
```
